optaplanner-core/src/test/java/org/optaplanner/core/impl/domain/common/accessor/lambda/PropertySetterAccessorTest.java

```diff
@@ -0,0 +1,126 @@
+    public static Collection<Object[]> data() {
+        Collection<Object[]> params = new ArrayList<>();
+        params.add(new Object[]{"integerProperty", 5});
+        params.add(new Object[]{"shortProperty", (short) 5});
+        params.add(new Object[]{"byteProperty", (byte) 5});
+        params.add(new Object[]{"booleanProperty", true});
+        params.add(new Object[]{"floatProperty", 5.5f});
+        params.add(new Object[]{"doubleProperty", 5.5});
+        params.add(new Object[]{"objectProperty", "test"});
+        params.add(new Object[]{"charProperty", 'a'});
+        params.add(new Object[]{"arrayProperty", new char[]{'a', 'b'}});
+
+        return params;
+    }
+
+    @Test
+    public void testSetterAccessors() throws NoSuchFieldException, IllegalAccessException {
+        Class<?> declaringClass = MockClassWithSetters.class;
+        Field field = declaringClass.getDeclaredField(propertyName);
+        Class<?> propertyType = field.getType();
+
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        Method setterMethod = ReflectionHelper.getSetterMethod(declaringClass, propertyType, propertyName);
+        PropertySetterAccessor<?> propertySetterAccessor =
+                PropertySetterAccessor.create(setterMethod, propertyType, lookup);
+        assertNotNull(propertySetterAccessor);
+
+        MockClassWithSetters mockBean = new MockClassWithSetters();
+        propertySetterAccessor.executeSetter(mockBean, value);
+
+        assertEquals(value, field.get(mockBean));
+    }
+
+    static class MockClassWithSetters {
+
+        int integerProperty;
+        boolean booleanProperty;
+        String objectProperty;
+        double doubleProperty;
+        float floatProperty;
+        short shortProperty;
+        char charProperty;
+        byte byteProperty;
+        char[] arrayProperty;
+
+        public void setIntegerProperty(final int integerProperty) {
+            this.integerProperty = integerProperty;
+        }
+
+        public void setBooleanProperty(final boolean booleanProperty) {
+            this.booleanProperty = booleanProperty;
+        }
+
+        public void setObjectProperty(final String objectProperty) {
+            this.objectProperty = objectProperty;
+        }
+
+        public void setDoubleProperty(final double doubleProperty) {
+            this.doubleProperty = doubleProperty;
+        }
+
+        public void setFloatProperty(final float floatProperty) {
+            this.floatProperty = floatProperty;
+        }
+
+        public void setShortProperty(final short shortProperty) {
+            this.shortProperty = shortProperty;
+        }
+
+        public void setCharProperty(final char charProperty) {
+            this.charProperty = charProperty;
+        }
+
+        public void setByteProperty(final byte byteProperty) {
+            this.byteProperty = byteProperty;
+        }
+
+        public void setArrayProperty(final char[] arrayProperty) {
```

Comment:
We don't use the "final" keyword before method parameters normally. So for consistency with the rest of the code, would you mind removing it?

Reply:
For the sake of consistency, ok, will do